src/controlPanel/data/datasetsWidget.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+
+import { CommandRegistry } from '@lumino/commands';
+import { JSONObject } from '@lumino/coreutils';
+import { Menu } from '@lumino/widgets';
+
+import { ReactWidget } from '@jupyterlab/ui-components';
+
+import { IControlPanelModel, IGlueSessionSharedModel } from '../../types';
+
+export class DatasetsWidget extends ReactWidget {
+  constructor(options: {
+    model: IControlPanelModel;
+    commands: CommandRegistry;
+  }) {
+    super();
+    const { model, commands } = options;
+
+    this._model = model;
+    this.title.label = 'Datasets';
+
+    this._model.glueSessionChanged.connect(this._sessionChanged, this);
+
+    // Create a context menu.
+    this._menu = new Menu({ commands });
+    this._menu.addItem({ command: 'new-viewer' });
+
+    this._model.selectedDatasetChanged.connect(this.update, this);
+  }
+
+  private _contextMenu(event: React.MouseEvent) {
+    const target = event.target as HTMLElement;
+    if (this._dataNames.includes(target.id)) {
+      this._model.selectedDataset = target.id;
+    }
+
+    event.preventDefault();
+    const x = event.clientX;
+    const y = event.clientY;
+    this._menu.open(x, y);
+  }
+
+  private _sessionChanged() {
+    if (this._currentSharedModel) {
+      this._currentSharedModel.contentsChanged.disconnect(
+        this._updateDataSets,
+        this
+      );
+    }
+
+    if (!this._model.sharedModel) {
+      this._currentSharedModel = null;
+      return;
+    }
+
+    this._currentSharedModel = this._model.sharedModel;
+    this._updateDataSets();
+    this._currentSharedModel.contentsChanged.connect(
+      this._updateDataSets,
+      this
+    );
+  }
+
+  private _updateDataSets() {
+    if (!this._currentSharedModel) {
+      return;
+    }
+
+    const mainModel = this._currentSharedModel.contents
+      .__main__ as JSONObject | null;
+
+    this._currentSharedModel.changed.connect(this._updateDataSets, this);
+
+    if (!mainModel?.data) {
+      return;
+    }
+
+    const dataCollection =
+      this._currentSharedModel.contents[mainModel.data as string];
+
+    this._dataNames = (dataCollection as JSONObject | null)?.data as string[];
+    this.update();
+  }
+
+  private _onClick(event: React.MouseEvent) {
+    const target = event.target as HTMLElement;
+    if (this._dataNames.includes(target.id)) {
+      this._model.selectedDataset = target.id;
+    }
+  }
+
+  render() {
```

Comment:
To avoid lint warning :

```suggestion
  render(): JSX.Element {
```